core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Manager.java

```diff
@@ -81,25 +85,25 @@ public OAuth2Util.AuthSession initSession(RESTClient initClient, Map<String, Str
       this.startTimeMillis = System.currentTimeMillis();
       this.authResponse =
           OAuth2Util.fetchToken(
-              initClient,
-              headers,
+              initClient.withAuthSession(session),
+              Map.of(),
               config.credential(),
               config.scope(),
               config.oauth2ServerUri(),
               config.optionalOAuthParams());
       return OAuth2Util.AuthSession.fromTokenResponse(
-          initClient, null, authResponse, startTimeMillis, session);
+          null, null, authResponse, startTimeMillis, session);
```

Comment:
can you elaborate why we're not passing the client here and a few lines below anymore? I understand that in both case we're not doing a refresh but I would probably still just pass a non-null client